src/exports-for-base/StateMachine.ts

```diff
@@ -1,20 +1,28 @@
 const globalMap = new WeakMap();
+const proxyMap = new WeakMap();
```

Comment:
What is the use case for proxyMap? I see that it's used for indexers, but why is this needed in addition to globalMap?

Reply:
When you use awaitedhandler in the classes, Internal access to "this" was not finding itself in the globalmap. But if you get state for the object from "outside" the class, it seems to be using the proxy object "identity". To be honest, I don't quite understand the behavior I was seeing here, but it's likely because the constructor returns a Proxy to the outside world.